pkg/operator/sync_common.go

```diff
@@ -262,3 +230,23 @@ func manageDeployment(client appsclientv1.AppsV1Interface, eventRecorder events.
 func serviceServingCertSignerName() string {
 	return fmt.Sprintf("%s@%d", "openshift-service-serving-signer", time.Now().Unix())
 }
+
+// cleanupDeprecatedResources ensures the deletion of resources no longer required by the operator.
+func cleanupDeprecatedResources(c serviceCAOperator) error {
```

Comment:
while fine here, this could become a cleanup helper for library-go using the dynamic client. Wdyt?

Reply:
Do you mean allowing a simple way to declare resources that are to be removed, as follows?

```golang
type DeletionTarget { 
  Object pkgruntime.Object
  Name string
  Namespace string
}

targets := []DeletionTargets{
  {v1.ClusterRole{}, "foo"},
  {myType{}, "bar", "foo"},
}
```

This should be sufficient input to a helper using the dynamic or controller-runtime client.

Or did you want to avoid typing altogether? As much as I'm happy dealing with GVK or GVR, I'm not sure that's nice UX for most.

Reply:
we have `Coordinate`s for arbitrary resource I think.

Reply:
I see `ResourceLocation`, which can be used for name+namespace, is that what you meant? Is there something similar to describe type details? I'm not sure how it could be easier than an obj or gvr, though, given that there is inherent complexity in construing the rest url.